HW0/P6/P6B.py

```diff
@@ -1,36 +1,43 @@
 import multiprocessing as mp
+import numpy as np
 import time
 import matplotlib.pyplot as plt
 
 # Sleep for t seconds
 def burnTime(t):
-    time.sleep(t)
+	time.sleep(t)
 
 # Main
 if __name__ == '__main__':
-    N = 16  # The number of jobs
-    P = 4   # The number of processes
+	N = 16  # The number of jobs
+	P = 4   # The number of processes
 
     # A thread pool of P processes
-    pool = mp.Pool(P)
+	pool = mp.Pool(P)
 
     # Use a variety of wait times
-    ratio = []
-    wait_time = []
+	ratio = []
+	wait_time = [10**-6,5*10**-6,7.5*10**-6,10**-5,5*10**-5,7.5*10**-5,10**-4,5*10**-4,7.5*10**-4,10**-3,5*10**-3,7.5*10**-3,10**-2,5*10**-2,7.5*10**-2,10**-1,5*10**-1,7.5*10**-1,1]
 
-    for t in wait_time:
-        # Compute jobs serially and in parallel
-        # Use time.time() to compute the elapsed time for each
-        serialTime = 1
-        parallelTime = 1
+	for t in wait_time:
+    	# Compute jobs serially and in parallel
+		# Use time.time() to compute the elapsed time for each
+		before = time.time()
+		for i in range(16):
+			burnTime(t)
+		serialTime = time.time()-before
+		
+		before = time.time()
+		pool.map(burnTime(t), range(N))
```

Comment:
The use of pool.map is indeed incorrect. If you look at the graph, you can see that you get a speedup that is higher than 4 (the number of processes).